src/render/src/pso/buffer.rs

```diff
@@ -132,13 +136,18 @@ impl<'a> DataLink<'a> for RawVertexBuffer {
 
 impl<R: Resources> DataBind<R> for RawVertexBuffer {
     type Data = handle::RawBuffer<R>;
-    fn bind_to(&self, out: &mut RawDataSet<R>, data: &Self::Data, man: &mut handle::Manager<R>) {
+    fn bind_to(&self,
+               out: &mut RawDataSet<R>,
+               data: &Self::Data,
+               man: &mut handle::Manager<R>,
+               access: &mut AccessInfo<R>) {
         let value = Some((man.ref_buffer(data).clone(), 0));
         for i in 0 .. MAX_VERTEX_ATTRIBUTES {
             if (self.1 & (1<<i)) != 0 {
                 out.vertex_buffers.0[i] = value;
             }
         }
+        access.buffer_read(data);
```

Comment:
I think this line (and the whole function body) should be wrapped into `if self.1 != 0` condition in order to avoid any actions for unused buffers.